src/sentry/toolbar/views/iframe_view.py

```diff
@@ -1,74 +1,91 @@
 from typing import Any
 
-from django.http import HttpRequest, HttpResponse
+from django.http import HttpRequest
 
 from sentry.models.organization import Organization
 from sentry.models.project import Project
-from sentry.toolbar.utils.url import check_origin
-from sentry.web.frontend.base import OrganizationView, region_silo_view
+from sentry.toolbar.utils.url import is_origin_allowed
+from sentry.web.frontend.base import ProjectView, region_silo_view
 
-REFERRER_HEADER = "HTTP_REFERER"  # 1 R is the spelling used here: https://docs.djangoproject.com/en/5.1/ref/request-response/
-SUCCESS_TEMPLATE = "sentry/toolbar/iframe.html"
-INVALID_TEMPLATE = "sentry/toolbar/iframe-invalid.html"
+TEMPLATE = "sentry/toolbar/iframe.html"
+
+
+def _get_referrer(request) -> str | None:
+    # 1 R is because of legacy http reasons: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer
+    return request.META.get("HTTP_REFERER")
 
 
 @region_silo_view
-class IframeView(OrganizationView):
-    def respond(self, template: str, context: dict[str, Any] | None = None, status: int = 200):
+class IframeView(ProjectView):
+    def respond(self, template: str, context: dict[str, Any], status: int = 200):
```

Comment:
```suggestion
    def respond(self, template: str, context: dict[str, Any] | None = None, status: int = 200):
```
Nit: match signature of superclass